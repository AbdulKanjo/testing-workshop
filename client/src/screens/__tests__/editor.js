import React from 'react'
import {generate, render, Simulate, flushPromises} from 'client-test-utils'
import Editor from '../editor'

test('calls onSubmit with the username and password when submitted', async () => {
  // Arrange
  const fakeUser = generate.userData({id: generate.id()})
  const fakePost = generate.postData({authorId: fakeUser.id})
  const fakeHistory = {push: jest.fn()}
  const fakeApi = {
    posts: {
      create: jest.fn(() => Promise.resolve()),
    },
  }
  const {queryByTestId} = render(
    <Editor api={fakeApi} user={fakeUser} history={fakeHistory} />,
  )

  queryByTestId('title-input').value = fakePost.title
  queryByTestId('content-input').value = fakePost.content
  queryByTestId('tags-input').value = fakePost.tags.join(', ')
  const submitButtonNode = queryByTestId('editor-submit')
  const formNode = queryByTestId('editor-form')
  const preDate = Date.now()

  // Act
  Simulate.submit(formNode)

  const postDate = Date.now()
  await flushPromises()

  // Assert
  expect(fakeApi.posts.create).toHaveBeenCalledTimes(1)
  expect(fakeApi.posts.create).toHaveBeenCalledWith({
    ...fakePost,
    date: expect.any(String),
  })
  const date = new Date(fakeApi.posts.create.mock.calls[0][0].date).getTime()
  expect(date).toBeGreaterThanOrEqual(preDate)
  expect(date).toBeLessThanOrEqual(postDate)
  expect(submitButtonNode.type).toBe('submit')
})

test('snapshot', () => {
  const {container} = render(<Editor />)
  expect(container.firstChild).toMatchSnapshot()
})
