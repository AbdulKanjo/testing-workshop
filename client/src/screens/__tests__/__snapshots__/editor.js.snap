// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
.css-0,
[data-css-0] {
  background: white;
  height: 50px;
  border: none;
  border-radius: 10px;
  box-shadow: var(--shadow);
  border-bottom: 5px;
  width: 30%;
  display: block;
  margin: 0 auto 10px auto;
  padding-left: 10px;
}

.css-0::-webkit-input-placeholder,
[data-css-0]::-webkit-input-placeholder {
  opacity: 0.5;
}

.css-0::-moz-placeholder,
[data-css-0]::-moz-placeholder {
  opacity: 0.5;
}

.css-0::-ms-input-placeholder,
[data-css-0]::-ms-input-placeholder {
  opacity: 0.5;
}

.css-0::placeholder,
[data-css-0]::placeholder {
  opacity: 0.5;
}

.css-5,
[data-css-5] {
  background: white;
  border: none;
  height: 200px;
  border-radius: 10px;
  box-shadow: var(--shadow);
  border-bottom: 5px;
  width: 30%;
  display: block;
  margin: 0 auto 10px auto;
  padding-top: 10px;
  padding-left: 10px;
}

.css-5::-webkit-input-placeholder,
[data-css-5]::-webkit-input-placeholder {
  opacity: 0.5;
}

.css-5::-moz-placeholder,
[data-css-5]::-moz-placeholder {
  opacity: 0.5;
}

.css-5::-ms-input-placeholder,
[data-css-5]::-ms-input-placeholder {
  opacity: 0.5;
}

.css-5::placeholder,
[data-css-5]::placeholder {
  opacity: 0.5;
}

.css-10,
[data-css-10] {
  background: var(--green);
  padding: 10px 20px;
  display: block;
  margin: auto;
  color: white;
  border: none;
  border-radius: 10px;
  box-shadow: var(--shadow);
  cursor: pointer;
  transition: 0.5s;
  -webkit-transition: 0.5s;
  -moz-transition: 0.5s;
}

.css-10:hover,
[data-css-10]:hover {
  box-shadow: var(--shadowHover);
}

@media only screen and (max-width: 744px) {
  .css-0,
  [data-css-0] {
    width: 70%;
  }
}

@media only screen and (max-width: 744px) {
  .css-5,
  [data-css-5] {
    width: 70%;
  }
}

<div>
  <form
    data-test="editor-form"
  >
    <input
      class="css-0"
      data-test="title-input"
      name="title"
      placeholder="title"
    />
    <textarea
      class="css-5"
      data-test="content-input"
      name="content"
      placeholder="content"
    />
    <input
      class="css-0"
      data-test="tags-input"
      name="tags"
      placeholder="tags"
    />
    <input
      class="css-10"
      data-test="editor-submit"
      type="submit"
    />
  </form>
</div>
`;
